chain/client/src/stateless_validation/partial_witness/partial_witness_tracker.rs

```diff
@@ -43,7 +43,7 @@ enum AccessedContractsState {
     Unknown,
     /// Received `ChunkContractAccesses` and sent `ContractCodeRequest`,
     /// waiting for response from the chunk producer.
-    Requested { contract_hashes: BTreeSet<CodeHash>, requested_at: Instant },
+    Requested { contract_hashes: HashSet<CodeHash>, requested_at: Instant },
```

Comment:
What're the pros and cons of BTreeSet vs HashSet here? Why are we using one vs the other? Would Vec<CodeHash> also work?

Reply:
The reason for set vs vector is to indicate using the type that the items are unique.

HashSet is known to be faster than BTreeSet and also since we do not need order/sortedness, thus I wanted to replace it with HashSet.